### PR TITLE
fix: sort tags by name

### PR DESCRIPTION
**Pull request by blizzz**
Alternative to #38066

@tobiasKaminsky please test, i did only superficially.

### CODE REVIEW

**Review by icewind1991 — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Review by AndyScherzinger — APPROVED** [review]

**Comment by tobiasKaminsky** [comment]
Tested and works, api wise.

File: apps/dav/lib/SystemTag/SystemTagPlugin.php [hunk 1]
```diff
@@ -306,8 +308,11 @@ private function propfindForFile(PropFind $propFind, Node $node): void {
 			if ($user === null) {
 				return;
 			}
-	
+
 			$tags = $this->getTagsForFile($node->getId(), $user);
+			usort($tags, function (SystemTag $tagA, SystemTag $tagB): int {
```
Comment (come-nc) [inline apps/dav/lib/SystemTag/SystemTagPlugin.php:313]:
```suggestion
			usort($tags, function (ISystemTag $tagA, ISystemTag $tagB): int {
```
The interface has the `getName` method, why type this to the internal implementation?

File: apps/dav/lib/SystemTag/SystemTagPlugin.php [hunk 2]
```diff
@@ -25,6 +25,7 @@
  */
 namespace OCA\DAV\SystemTag;
 
+use OC\SystemTag\SystemTag;
```
Comment (come-nc) [inline apps/dav/lib/SystemTag/SystemTagPlugin.php:28]:
```suggestion
```
Apps should not use `OC`. I do not see the need here.

**Review by come-nc — DISMISSED** [review]

**Comment by tobiasKaminsky** [comment]
When adding a tag "0123", then it first is shown correctly.
image

But on page refresh it is like this:
image

**Review by tobiasKaminsky — DISMISSED** [review]
see comment about "0123"

**Comment by blizzz** [comment]
@tobiasKaminsky I can reproduce this behaviour with current master, so these changes are not causing it. Please open a separate issue.

**Pull request merged**